util/lua/lua.go

```diff
@@ -275,48 +275,60 @@ func cleanReturnedArray(newObj, obj []interface{}) []interface{} {
 	return arrayToReturn
 }
 
-func (vm VM) ExecuteResourceActionDiscovery(obj *unstructured.Unstructured, script string) ([]appv1.ResourceAction, error) {
-	l, err := vm.runLua(obj, script)
-	if err != nil {
-		return nil, err
+func (vm VM) ExecuteResourceActionDiscovery(obj *unstructured.Unstructured, scripts []string) ([]appv1.ResourceAction, error) {
+	if len(scripts) == 0 {
+		return nil, fmt.Errorf("no action discovery script provided")
 	}
-	returnValue := l.Get(-1)
-	if returnValue.Type() == lua.LTTable {
-		jsonBytes, err := luajson.Encode(returnValue)
-		if err != nil {
-			return nil, err
-		}
-		availableActions := make([]appv1.ResourceAction, 0)
-		if noAvailableActions(jsonBytes) {
-			return availableActions, nil
-		}
-		availableActionsMap := make(map[string]interface{})
-		err = json.Unmarshal(jsonBytes, &availableActionsMap)
+	availableActions := make([]appv1.ResourceAction, 0)
+
+	for _, script := range scripts {
+		l, err := vm.runLua(obj, script)
 		if err != nil {
 			return nil, err
 		}
-		for key := range availableActionsMap {
-			value := availableActionsMap[key]
-			resourceAction := appv1.ResourceAction{Name: key, Disabled: isActionDisabled(value)}
-			if emptyResourceActionFromLua(value) {
-				availableActions = append(availableActions, resourceAction)
-				continue
+		returnValue := l.Get(-1)
+		if returnValue.Type() == lua.LTTable {
+			jsonBytes, err := luajson.Encode(returnValue)
+			if err != nil {
+				return nil, fmt.Errorf("error in converting to lua table: %w", err)
 			}
-			resourceActionBytes, err := json.Marshal(value)
+			if noAvailableActions(jsonBytes) {
+				return availableActions, nil
+			}
+			availableActionsMap := make(map[string]interface{})
+			err = json.Unmarshal(jsonBytes, &availableActionsMap)
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("error in converting to lua table: %w", err)
 			}
+			for key, value := range availableActionsMap {
+				resourceAction := appv1.ResourceAction{Name: key, Disabled: isActionDisabled(value)}
+				if emptyResourceActionFromLua(value) {
+					availableActions = append(availableActions, resourceAction)
+					continue
+				}
+				if getUniqActions(resourceAction.Name, availableActions) {
```

Comment:
instead of iterating the array, i'm using map to check if key exist
https://github.com/argoproj/argo-cd/pull/19708/files#diff-c22c83d4dc2896d3997ee7131fb88693ddfec7e36ca35aac63aa8b0f0f057bd5R305